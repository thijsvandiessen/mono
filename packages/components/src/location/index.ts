export { Location } from './location.jsx'
export { LocationView } from './locationView.jsx'
