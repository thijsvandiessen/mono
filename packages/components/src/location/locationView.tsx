import type { Location } from '@mono/datocms'
import React from 'react'

export interface Props {
  data: Location
}

export const LocationView = ({ data }: Props) => {
  const { title, address, startTime } = data
  return (
    <address>
      <h2>{title}</h2>
      {address && <p>{address}</p>}
      {startTime && <p>{new Date(startTime).toDateString()}</p>}
    </address>
  )
}
