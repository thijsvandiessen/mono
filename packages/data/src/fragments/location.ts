import { type TypedDocumentNode, gql } from '@urql/core'
import type { LocationFragment } from '../generated/graphql.js'
import { coordinates } from './coordinates.js'

export const location: TypedDocumentNode<LocationFragment> = gql`
  fragment location on LocationRecord {
    id
    title
    addressTitle
    address {
      ...coordinates
    }
  }
  ${coordinates}
`
