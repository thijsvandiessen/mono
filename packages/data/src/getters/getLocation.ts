import {
  GetLocationDocument,
  type GetLocationQuery,
  type GetLocationQueryVariables,
} from '../generated/graphql.js'
import type { CombinedError } from '@urql/core'
import type { Location } from '../types/location.js'
import { client } from '../gqlClient.js'
import { locationFormatter } from '../formatters/locationFormatter.js'

export const getLocation = async ({
  id,
}: GetLocationQueryVariables): Promise<
  | {
      data: Location | null
      error?: CombinedError
    }
  | {
      data: null
      error: unknown
    }
> => {
  try {
    const { data, error } = await client.query<
      GetLocationQuery,
      GetLocationQueryVariables
    >(GetLocationDocument, {
      id,
    })

    return {
      data: data?.location?.id ? locationFormatter(data.location) : null,
      error,
    }
  } catch (error) {
    if (error instanceof Error) console.log(error.message)
    return { data: null, error }
  }
}
