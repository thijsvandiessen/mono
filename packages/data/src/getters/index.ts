export { getConcert } from './getConcert.js'
export { getConcertPage } from './getConcertPage.js'
export { getConcertPageSeo } from './getConcertPageSeo.js'
export { getConcerts } from './getConcerts.js'
export { getConcertsMeta } from './getConcertsMeta.js'
export { getConcertUrls } from './getConcertUrls.js'
export { getFutureConcerts } from './getFutureConcerts.js'
export { getGeneralInfo } from './getGeneralInfo.js'
export { getLocation } from './getLocation.js'
export { getLocations } from './getLocations.js'
export { getPage } from './getPage.js'
export { getPages } from './getPages.js'
export { getPageSeo } from './getPageSeo.js'
export { getPagesMeta } from './getPagesMeta.js'
export { getSiteInfo } from './getSiteInfo.js'
export { getSiteMetadata } from './getSiteMetadata.js'
