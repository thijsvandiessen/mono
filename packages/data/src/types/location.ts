export interface Location {
  id: string
  title: string | null
  address: string | null
  lat: number | null
  lng: number | null
  startTime: string | null
  ticketLink: string | null
}
