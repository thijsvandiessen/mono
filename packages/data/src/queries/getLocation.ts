import type {
  GetLocationQuery,
  GetLocationQueryVariables,
} from '../generated/graphql.js'
import { type TypedDocumentNode, gql } from '@urql/core'
import { location } from '../fragments/location.js'

export const GET_LOCATION: TypedDocumentNode<
  GetLocationQuery,
  GetLocationQueryVariables
> = gql`
  query getLocation($id: ItemId) {
    location(filter: { id: { eq: $id } }) {
      ...location
    }
  }
  ${location}
`
