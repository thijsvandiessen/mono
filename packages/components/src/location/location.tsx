import { LocationView } from './locationView'
import React from 'react'
import { getLocation } from '@mono/datocms'

export interface Props {
  id: string
}

export const Location = async ({ id }: Props) => {
  const { data } = await getLocation({ id })
  if (!data) return null

  return <LocationView data={data} />
}
