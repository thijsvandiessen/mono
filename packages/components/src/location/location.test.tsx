import React from 'react'
import { Location } from './location'
import type { Props } from './location'
import { getLocation } from '@mono/graphql/src/getters/getLocation'
import { resolvedComponent } from '@mono/utils'
import { render, screen } from '@testing-library/react'

jest.mock('@mono/graphql/src/getters/getLocation', () => {
  const originalModule = jest.requireActual(
    '@mono/graphql/src/getters/getLocation'
  )
  return {
    __esModule: true,
    ...originalModule,
    getLocation: jest.fn(),
  }
})

const getLocationMock = jest.mocked(getLocation)

describe('Concert component', () => {
  it('shows all the data', async () => {
    getLocationMock.mockResolvedValueOnce({
      data: {
        id: 'mock-ids',
        title: 'title',
        address: 'some address lines',
        lat: 5,
        lng: 42,
      },
      error: undefined,
    })

    const Resolved = await resolvedComponent<Props>(Location, {
      id: 'mock-id',
    })

    const { container } = render(<Resolved />)

    expect(container).toMatchSnapshot()
    expect(screen.getByText('title')).toBeTruthy()
  })
})
