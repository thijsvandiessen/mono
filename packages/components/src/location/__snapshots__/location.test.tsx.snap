// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Location component > shows all the data 1`] = `
<div>
  <address>
    <h2>
      title
    </h2>
    <p>
      some address lines
    </p>
    <p>
      Fri Sep 01 2023
    </p>
  </address>
</div>
`;
